datacompy/spark.py

```diff
@@ -13,25 +13,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
 import sys
 from enum import Enum
 from itertools import chain
+from typing import Any, TextIO, NewType, List, Union, Tuple, Optional, Dict, Set
+
+import pyspark
```

Comment:
@jdawang @aguiddir @Jin66  I think this breaks the below logic of not mandating pyspark? I might be wrong, but if you don't have pyspark installed, this will still get executed by the `__init__` and throw an `ImportError`

Reply:
In https://github.com/capitalone/datacompy/blob/3b078e6666a253a96298e1a372fbf419ed6b975a/datacompy/__init__.py#L27

This will raise `ModuleNotFound` error on `import datacompy` if pyspark is a top-level import instead of within the try-except block.

Reply:
If you `import datacompy`, [__init__](https://github.com/capitalone/datacompy/blob/develop/datacompy/__init__.py) is called. In particular:
```python
from datacompy.spark import NUMERIC_SPARK_TYPES, SparkCompare
```
This happens even if you have no intention of using `SparkCompare`. Sorry for not being clearer. Not sure if this make sense, but the `import pyspark` will get executed and if you don't have it installed it will complain in this case. this is why right below we have the `try`, `except`

Reply:
I just tested for my own sanity:

```python
>>> import datacompy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[REDACTED]/datacompy/__init__.py", line 27, in <module>
    from datacompy.spark import NUMERIC_SPARK_TYPES, SparkCompare
  File "[REDACTED]/datacompy/spark.py", line 21, in <module>
    import pyspark
ModuleNotFoundError: No module named 'pyspark'
```

Reply:
@jdawang I like your idea of adding in a test without installing Spark and running the tests

Reply:
@fdosani, I made the changes to use string version instead of object for typing. Thanks for the suggestion